### PR TITLE
Update nvFuser matmul

### PR DESCRIPTION
**Pull request by Priya2698**
Following https://github.com/NVIDIA/Fuser/pull/2175 and https://github.com/NVIDIA/Fuser/pull/2209, nvFuser supports all cases for matmul as `torch.matmul`. This PR removes the previous restrictions and updates the tests.

### CODE REVIEW

File: thunder/executors/nvfuserex_impl.py [hunk 1]
```diff
@@ -2236,17 +2236,11 @@ def _matmul_check(
     a: TensorProxy,
     b: TensorProxy,
 ) -> bool:
-    if nv_version < LooseVersion("0.2.2"):
+    if nv_version < LooseVersion("0.2.4"):
```
[the remark this answers is not in the document]
Comment (Priya2698) [reply thunder/executors/nvfuserex_impl.py:2239]: That version of `matmul` had limited functionality due to segmentation issues seen around it (see https://github.com/NVIDIA/Fuser/issues/2127) which are resolved in the updated version. So we should be using 0.2.4 for `matmul` preferably.

I don't see much value in including 0.2.2 rn.

OOC, do we test against intermediate nvFuser versions anywhere? The CI uses the PyPI version (0.2.0) and the latest branch.

[the remark this answers is not in the document]
Comment (crcrpar) [reply thunder/executors/nvfuserex_impl.py:2239]: Right but it technically can be a breaking change for those who somehow wants to pin nvfuser to 0.2.2 by losing the opportunity of nvfuser matmuls

[the remark this answers is not in the document]
Comment (Priya2698) [reply thunder/executors/nvfuserex_impl.py]: Yes, that's possible. I included it with a warning for previous version.

**Review by crcrpar — APPROVED** [review]

**Review by t-vi — APPROVED** [review]
Thank you @Priya2698 @crcrpar

**Pull request merged**